src/views/PickupLocationModal.vue

```diff
@@ -0,0 +1,182 @@
+      <ion-item>
+        <ion-list>
+          <ion-label>{{ shipGroup.shipTo.postalAddress.toName }}</ion-label>
+          <ion-label color="dark">{{ shipGroup.shipTo.postalAddress.address1 }} </ion-label>
+          <ion-label color="dark">{{ shipGroup.shipTo.postalAddress.city }} {{ shipGroup.shipTo.postalAddress.country }} {{ shipGroup.shipTo.postalAddress.postalCode }}</ion-label>
+        </ion-list>
+      </ion-item>
+    </ion-card>
+    <ion-radio-group v-if="nearbyStores.length" v-model="facilityId">
+      <ion-list>
+        <ion-list-header lines="full" color="light">
+          <ion-label>{{ $t("Nearby stores") }}</ion-label>
+        </ion-list-header>
+        <ion-item v-for="store of nearbyStores" :key="store.facilityId">
+          <ion-label>{{ store.facilityName }}</ion-label>
+          <!-- Showing store distance in miles -->
+          <ion-label v-if="store.distance" slot="end">{{ store.distance }} {{ $t("mi") }}</ion-label>
+          <ion-radio :value="store.facilityId" slot="end" />
+        </ion-item>
+      </ion-list>
+    </ion-radio-group>
+    <ion-item v-else lines="none" class="ion-text-center">
+      <ion-label>{{ $t("No nearby stores found") }}</ion-label>
+    </ion-item>
+    <!-- Only show select button if there are stores to select from -->
+    <div v-if="nearbyStores.length" class="ion-text-center">
+      <ion-button @click="updateShipmentStore()">{{ $t("Select pickup location") }}</ion-button>
+    </div>
+  </ion-content>
+</template>
+
+<script lang="ts">
+import {
+  IonButton,
+  IonButtons,
+  IonCard,
+  IonContent,
+  IonHeader,
+  IonIcon,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonListHeader,
+  IonRadio,
+  IonRadioGroup,
+  IonTitle,
+  IonToolbar,
+  modalController
+} from '@ionic/vue';
+import { defineComponent } from 'vue';
+import { closeOutline } from 'ionicons/icons';
+import { useRouter } from "vue-router";
+import { useStore } from "@/store";
+import { FacilityService } from '@/services/FacilityService';
+import { StockService } from '@/services/StockService';
+import { translate } from '@/i18n';
+import { hasError, showToast } from '@/utils';
+
+export default defineComponent({
+  name: 'StoreModal',
+  components: {
+    IonButton,
+    IonButtons,
+    IonCard,
+    IonContent,
+    IonHeader,
+    IonIcon,
+    IonItem,
+    IonList,
+    IonListHeader,
+    IonLabel,
+    IonRadio,
+    IonRadioGroup,
+    IonTitle,
+    IonToolbar
+  },
+  data() {
+    return {
+      nearbyStores: [] as any,
+      facilityId: ''
+    }
+  },
+  props: ["shipGroup", "productIds"],
+  mounted() {
+    this.getNearbyStores()
+  },
+  methods: {
+    async getNearbyStores() {
+      try {
+        let shipGroupLocationResp: any, productInventoryResp: any, storeLookupResp: any, facilityIds;
+        if (this.shipGroup.shipmentMethodTypeId === 'STOREPICKUP' && this.shipGroup.selectedShipmentMethodTypeId === 'STOREPICKUP') {
+          storeLookupResp = await FacilityService.getStores({
```

Comment:
Create separate method for each specific use case, like for fetching stores, for checking inventory and for getting latLon for zipCode.

Reply:
Done sir